File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        int answer = 0;
        int wMax = 0;
        int hMax = 0;

        for (int i = 0; i < sizes.length; i++) {
            int max = Math.max(sizes[i][0], sizes[i][1]);
            int min = Math.min(sizes[i][0], sizes[i][1]);
            wMax = Math.max(wMax, max);
            hMax = Math.max(hMax, min);
        }

        answer = wMax * hMax;

        return answer;
    }
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {
    let width = 0;
    let length = 0;
    sizes.forEach(s => {
        const [a, b] = s.sort((a, b) => a - b);
        width = Math.max(a, width);
        length = Math.max(b, length);
    });
    return width * length;
}